### ISSUE TITLE
Does not update message list after an image is sent

### ISSUE THREAD

**Issue by karabatov** [smooch/smooch-ios#100]
Not sure if it's a bug, but I haven't found any force-update message methods, so probably it is.

What I'm doing:
1. Sending the “Uploading screenshot…” message which appears instantly.
2. Uploading image and getting success and no error.
3. Nothing happens in the message list *until* I send another message, which appears along with the uploaded image at the same time.

As a workaround, I send the “Screenshot uploaded.” message right after I get a success, and both the image and the second message appear instantly.

Relevant bit of code:

```
let message = SKTMessage(text: "Uploading screenshot…")
Smooch.conversation().sendMessage(message)

Smooch.conversation().sendImage(image, withProgress: nil) { error, _ in
    guard let _ = error else {
        let message = SKTMessage(text: "Screenshot uploaded.")
        Smooch.conversation().sendMessage(message)
        return
    }

    // Log "Screenshot could not upload"
}
```

So my questions are:
1. Is it a bug?
2. Can I force-update the message list or insert the image message into the conversation somehow so that I don't have to send a followup message?

Thanks.

**Comment by mspensieri** [comment]
@karabatov The second parameter to the `completion` callback should be a reference to the newly created message, and at the time the block is invoked the message should already have been appended to the `messages` array. If it's not in fact in the array at that time, then this would definitely be a bug!

I'll try to reproduce on my side and let you know

**Comment by mspensieri** [comment]
Works correctly for me on the latest version:

```
[[Smooch conversation] sendImage:image withProgress:nil completion:^(NSError *error, SKTMessage *message) {
    NSLog(@"%@", message.messageId);
    NSLog(@"%@", [[Smooch conversation].messages containsObject:message] ? @"YES" : @"NO");
}];
```

Outputs:

```
2016-05-20 08:40:52.316 ShellApp[75783:4779933] 573f05d401fbbc4526d9b77a
2016-05-20 08:40:52.316 ShellApp[75783:4779933] YES
```

**Comment by karabatov** [comment]
@mspensieri I will check this on my side and get back to you.

**Comment by karabatov** [comment]
Sorry for being late on this, been pushing a release.

@mspensieri I did just like you did and here's what I found:

If I do something like this:
```
Smooch.conversation().sendImage(image, withProgress: nil) { [weak self] error, newMessage in
    print("Messages array contains image message:", (Smooch.conversation().messages as! [SKTMessage]).contains(newMessage))
}
```
It prints:
```

**Comment by mspensieri** [comment]
Hmm so you are using the `sendImage:` API while the conversation view is shown? Can you explain more your use case? Why is the user not using the built-in photo button?

**Comment by karabatov** [comment]
> so you are using the `sendImage:` API while the conversation view is shown?
Exactly. We are catching the event when user is making a screenshot and ask if they want to upload it. If they do, we grab the latest photo on their Camera Roll, open Smooch and send it away.

**Comment by mspensieri** [comment]
Ok I see what you mean. Yes, we don't properly handle this case. I'll file a bug internally and I'll close this issue when it has been fixed

**Comment by karabatov** [comment]
Cool! It's not critical, but a bit annoying :) Will be glad to see it fixed.

**Comment by mspensieri** [comment]
This should be fixed in [v4.3.0](https://github.com/smooch/smooch-ios/releases/tag/4.3.0)
